src/invidious/routes/before_all.cr

```diff
@@ -148,5 +148,11 @@ module Invidious::Routes::BeforeAll
     end
 
     env.set "current_page", URI.encode_www_form(current_page)
+
+    unregistered_path_whitelist = {"/", "/login", "/licenses", "/privacy"}
+    if !env.get?("user") && !unregistered_path_whitelist.includes?(env.request.path) && CONFIG.login_only
```

Comment:
don't know how crystal deals with this but its probably best to check the config flag first in order to not redundantly check this the whitelist on every request

Reply:
Ok, I have updated the patch with your suggestion, thanks.